src/Drivetrain.cpp

```diff
@@ -208,7 +214,7 @@ void DriveTrain::Loop()
 		}
 	}
 
-	if (m_isdownshifting && (abs(m_previousx * X_SCALING) < ENCODER_TOP_SPEED) && (abs(m_previousy * Y_SCALING) < ENCODER_TOP_SPEED))
+	if (m_isdownshifting && (abs(m_previousx * X_SCALING) < CHILD_PROOF_SPEED) && (abs(m_previousy * Y_SCALING) < CHILD_PROOF_SPEED))

```

Comment:
Several uses of abs on floating point values.  Unintentional truncation may be happening.  Suggest use of fabs or comment stating that integer truncation acceptable.